app/controllers/sessions_controller.rb

```diff
@@ -108,15 +108,32 @@ def auth_hash
     request.env["omniauth.auth"]
   end
 
-  def redirect_to_correct_page(user)
-    if user&.accept_terms_date_utc.nil?
+  def redirect_to_correct_page(user_session)
+    if user_session&.accept_terms_date_utc.nil?
       redirect_to accept_terms_path
-    elsif user&.state == "new"
-      redirect_to transition_info_path
-    elsif Settings.rollover && user&.state == "transitioned"
-      redirect_to rollover_path
+    elsif permitted_state
+      redirect_to redirect_path[permitted_state]
     else
       redirect_to session[:redirect_back_to] || root_path
     end
   end
+
+  def permitted_state
+    user = User.new(
```

Comment:
we need to initialise and instance of User in order to access `aasm`'s helper methods

Reply:
I think we have the user object already here, from line 27 in the `create` method:

```ruby
    user = set_user_session
```

When then gets passed to `redirect_to_correct_page` as `user_session`, so we shouldn't have to reify a user here I don't think. Maybe line 27 above should assign `@user`? We also do it on line 39 in `create_by_magic` so that may need to be `@user` too.

Seems like some tech debt from when we first implemented the sessions, can we have a look at how we might tidy this up a little bit?